Content.IntegrationTests/Tests/Roles/StartingGearStorageTests.cs

```diff
@@ -0,0 +1,64 @@
+using System.Linq;
+using System.Numerics;
+using Content.Shared.Roles;
+using Content.Server.Storage.EntitySystems;
+using Robust.Shared;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Collections;
+
+namespace Content.IntegrationTests.Tests.Roles;
+
+[TestFixture]
+public sealed class StartingGearPrototypeStorageTest
+{
+    /// <summary>
+    /// Checks that a storage fill on a StartingGearPrototype will properly fill
+    /// </summary>
+    [Test]
+    public async Task TestStartingGearStorage()
+    {
+        var settings = new PoolSettings { Connected = true, Dirty = true };
+        await using var pair = await PoolManager.GetServerClient(settings);
+        var server = pair.Server;
+        var mapManager = server.ResolveDependency<IMapManager>();
+        var storageSystem = server.System<StorageSystem>();
+
+        var protos = server.ProtoMan
+            .EnumeratePrototypes<StartingGearPrototype>()
+            .Where(p => !p.Abstract)
+            .ToList();
+
+        protos.Sort();
```

Comment:
you are not sorting simple ids anymore so you need to give it something to compare, something like
Sort( (x, y) => string.Compare(x.ID, y.ID, StringComparison.Ordinal) )

Reply:
another way to do the same would be a .OrderBy(p => p.ID)